src/librustc_mir/interpret/validation.rs

```diff
@@ -175,7 +175,7 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
                 }
             }
 
-            let result = self_.resolve_type_vars_if_possible(result);
+            let result = self_.resolve_type_and_region_vars_if_possible(result);
```

Comment:
Ah I didn't see that :/. Welp, that's what happens when there are leftover `ReErased` in the *input* to this.